RKNotificationHubTests/RKNotificationHubTests.m

```diff
@@ -1,40 +1,277 @@
 
-- (void)setUp {
-    [super setUp];
-    // Put setup code here. This method is called before the invocation of each test method in the class.
-}
+// setup
 
-- (void)tearDown {
-    // Put teardown code here. This method is called after the invocation of each test method in the class.
-    [super tearDown];
-}
+describe(@"initWithView:", ^{
+    __block UIView *view;
+    __block RKNotificationHub *subject;
+    
+    beforeAll(^{
+        view = [[UIView alloc] initWithFrame:CGRectZero];
+        subject = [[RKNotificationHub alloc] initWithView:view];
+    });
+    
+    it(@"should initialize hub with given view as hubView", ^{
+        expect(subject.hubView).to.equal(view);
+    });
+    
+    it(@"should initialize hub with count 0", ^{
+        expect([subject count]).to.equal(0);
+    });
+});
 
-- (void)testExample {
-    // This is an example of a functional test case.
-    XCTAssert(YES, @"Pass");
-}
+describe(@"setView:andCount:", ^{
+    __block UIView *view;
+    __block RKNotificationHub *subject;
+    
+    beforeAll(^{
+        view = [[UIView alloc] initWithFrame:CGRectZero];
+        subject = [[RKNotificationHub alloc] initWithView:nil];
+        [subject setView:view andCount:2];
+    });
+    
+    it(@"should set hub view", ^{
+        expect(subject.hubView).to.equal(view);
+    });
+    
+    it(@"should set count", ^{
+        expect([subject count]).to.equal(2);
+    });
+});
 
-- (void)testPerformanceExample {
-    // This is an example of a performance test case.
-    [self measureBlock:^{
-        // Put the code you want to measure the time of here.
-    }];
-}
+describe(@"changing the count", ^{
+    __block UIView *view;
+    __block RKNotificationHub *subject;
+    
+    before(^{
+        view = [[UIView alloc] initWithFrame:CGRectZero];
+        subject = [[RKNotificationHub alloc] initWithView:nil];
+        [subject setView:view andCount:20];
+    });
+    
+    describe(@"increment", ^{
+        it(@"should increase count", ^{
+            [subject increment];
+            expect([subject count]).to.equal(21);
+        });
+    });
+    
+    describe(@"incrementBy:", ^{
+        it(@"should increase count by given value", ^{
+            [subject incrementBy:10];
+            expect([subject count]).to.equal(30);
+        });
+    });
+    
+    describe(@"decrement", ^{
+        it(@"should decrease count", ^{
+            [subject decrement];
+            expect([subject count]).to.equal(19);
+        });
+        
+        it(@"should not decrease count if count is 0", ^{
+            [subject setCount:0];
+            [subject decrement];
+            expect([subject count]).to.equal(0);
+        });
+    });
+    
+    describe(@"decrementBy:", ^{
+        it(@"should decrease count by given value", ^{
+            [subject decrementBy:15];
+            expect([subject count]).to.equal(5);
+        });
+        
+        it(@"should make count negative if given value is larger than the count", ^{
```

Comment:
Yes, I've spotted this behaviour when I was playing with the library and decided to make it explicit by adding it to the spec. This example illustrates how specs can be useful for the end user (the developer, as you've stated earlier). Besides being a unit testing facility, the spec works like a documentation and a set of usage samples of the library and one can simply look through the spec to better understand its behaviour and spot not-so-obvious cases. As for this particular case, it shows that `decrement` and `decrementBy:` behave slightly different when current count is 0 (although I'm not sure that you have actually intended to have this distinction from the very beginning :) ).